packages/manager/modules/cloud-connect/src/cloud-connect.class.js

```diff
@@ -0,0 +1,176 @@
+  }
+
+  getAllowedPopType() {
+    if (this.isPopConfigurationExists()) {
+      return head(values(this.popConfiguration));
+    }
+    return null;
+  }
+
+  isPopConfigured(interfaceId) {
+    return (
+      !isEmpty(this.popConfiguration) && this.popConfiguration[interfaceId]
+    );
+  }
+
+  isPopConfigurationExists() {
+    return !isEmpty(this.popConfiguration);
+  }
+
+  isLoadingPop() {
+    return this.isLoadingPopConfiguration() || this.isLoadingInterface();
+  }
+
+  isLoadingPopConfiguration() {
+    return this.loadingPopConfiguration;
+  }
+
+  setPopConfiguration(configuration) {
+    if (configuration.interfaceId) {
+      this.popType = configuration.type;
+      this.popConfiguration[configuration.interfaceId] = new CloudConnectPop(
+        configuration,
+      );
+    }
+  }
+
+  addPopConfiguration(configuration) {
+    if (configuration.interfaceId) {
+      this.popType = configuration.type;
+      const popConfig = new CloudConnectPop(configuration);
+      this.popConfiguration[configuration.interfaceId] = popConfig;
+      return popConfig;
+    }
+    return false;
+  }
+
+  removePopConfiguration(interfaceId) {
+    if (interfaceId && this.popConfiguration[interfaceId]) {
+      this.popConfiguration[interfaceId] = null;
+      delete this.popConfiguration[interfaceId];
+    }
+  }
+
+  getPopConfiguration(interfaceId) {
+    return this.popConfiguration[interfaceId];
+  }
+
+  getFirstPopConfiguration() {
+    return this.getPopConfiguration(this.interfaceList[0]);
+  }
+
+  setLoadingPopConfiguration(loading) {
+    this.loadingPopConfiguration = loading;
+  }
+
+  setDcConfigurations(configurations) {
+    this.datacenterConfigurations = sortBy(configurations, ['dcName']) || [];
+  }
+
+  addDcConfiguration(dc) {
+    map(this.availableDatacenters, (datacenter) => {
+      if (datacenter.id === dc.datacenterId) {
+        set(datacenter, 'available', false);
+      }
+    });
+    this.datacenterConfigurations.push(dc);
+  }
+
+  removeDcConfiguration(dcId) {
+    remove(this.datacenterConfigurations, (dc) => dc.id === dcId);
+  }
+
+  getDcConfiguration(dcId) {
+    return find(this.datacenterConfigurations, (dc) => dc.id == dcId);
+  }
+
+  setLoadingServiceKeys(loading) {
+    this.loadingServiceKeys = loading;
+  }
+
+  isLoadingServiceKeys() {
+    return this.loadingServiceKeys;
+  }
+
+  setServiceKeys(serviceKeys) {
+    if (find(serviceKeys, { status: 'active' })) {
+      set(this, 'validServiceKey', true);
+    } else {
+      set(this, 'validServiceKey', false);
+    }
+    this.serviceKeys = serviceKeys;
+  }
+
+  getServiceKeys() {
+    return this.serviceKeys;
+  }
+
+  getActiveServiceKey() {
+    return find(
+      this.getServiceKeys(),
+      (key) => key.status === 'active' || key.status === 'doing',
+    );
+  }
+
+  setInterface(ccInterface) {
+    if (ccInterface.id) {
+      this.interfaces[ccInterface.id] = new CloudConnectInterface(ccInterface);
+    }
+  }
```

Comment:
I think `ccInterface` should be a `CloudConnectInterface` already and not be created by the `CloudConnect` class. But without types it's more complex to make sure the parameter is having the right type